internal/p2p/router_init_test.go

```diff
@@ -15,33 +15,42 @@ func TestRouter_ConstructQueueFactory(t *testing.T) {
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
 
+	getOpts := func() RouterOptions {
```

Comment:
Could this be a static value, or is it mutated during the test?

(If the latter, I'd suggest a comment)